internal/cluster/host_spec_subresource.go

```diff
@@ -0,0 +1,153 @@
+			"id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of a vSphere host in the free pool",
+			},
+			"host_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Host name of the vSphere host",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"availability_zone_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Availability Zone Name. This is required while performing a stretched cluster expand operation",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "IP address of the vSphere host",
+				ValidateFunc: validation_utils.ValidateIPv4AddressSchema,
+			},
+			"license_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "License key of a vSphere host in the free pool. This is required except in cases where the " +
+					"ESXi host has already been licensed outside of the VMware Cloud Foundation system",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"username": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Username of the vSphere host",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				Description:  "SSH password of the vSphere host",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"serial_number": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Serial Number of the vSphere hosts",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"ssh_thumbprint": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				Description:  "SSH thumbprint (fingerprint) of the vSphere host. Note: This field will be mandatory in future releases.",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"vmnic": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Contains vmnic configurations for vSphere host",
+				Elem:        network.VMNicSchema(),
+			},
+		},
+	}
+}
+
+func FlattenHost(host *models.HostReference) *map[string]interface{} {
+	result := make(map[string]interface{})
+	if host == nil {
+		return &result
+	}
+	result["id"] = host.ID
+	result["host_name"] = host.Fqdn
+	result["ip_address"] = host.IPAddress
+	result["availability_zone_name"] = host.AzName
+
+	return &result
+}
+
+func TryConvertToHostSpec(object map[string]interface{}) (*models.HostSpec, error) {
+	result := &models.HostSpec{}
+	if object == nil {
+		return nil, fmt.Errorf("cannot conver to HostSpec, object is nil")
+	}
+	id := object["id"].(string)
+	if len(id) == 0 {
+		return nil, fmt.Errorf("cannot conver to HostSpec, id is required")
+	}
+	result.ID = &id
+	if hostName, ok := object["host_name"]; ok && !validation_utils.IsEmpty(hostName) {
```

Comment:
Would need to be changes if `fqdn` is used.

Reply:
As stated above I'm sticking with "host_name" as, the way the API refers to this field.